sunpy/map/mapbase.py

```diff
@@ -2198,12 +2198,16 @@ def draw_contours(self, levels, axes=None, **contour_args):
         Notes
         -----
         Extra keyword arguments to this function are passed through to the
-        `~matplotlib.pyplot.contour` function.
+        `~matplotlib.axes.Axes.contour` function.
         """
         axes = self._check_axes(axes, allow_non_wcsaxes=True)
 
         levels = self._process_levels_arg(levels, warn_units=True)
-        cs = axes.contour(self.data, levels, **contour_args)
+        kwargs = {}
+        if wcsaxes_compat.is_wcsaxes(axes):
+            kwargs['transform'] = axes.get_transform(self.wcs)
+        kwargs.update(contour_args)
```

Comment:
Do we want to error if the user passes `transform` in `kwargs`?

Reply:
A power user might want to pass in a custom `transform` (e.g., when messing around with `RotatedSunFrame`!), so I don't think we should error or even warn.